rust/codegen/src/main.rs

```diff
@@ -0,0 +1,22 @@
+extern crate protoc_rust;
+
+use std::env;
+use std::vec::Vec;
+use protoc_rust::Customize;
+
+fn main() {
+	let args: Vec<_> = env::args().collect();
+	let mut root = "../..";
+	if args.len() > 1 {
+		root = &args[1];
+	}
+
+	protoc_rust::run(protoc_rust::Args {
+	    out_dir: &format!("{}{}", root, "/rust/src"),
+	    input: &[&format!("{}{}", root, "/proofs.proto")],
+	    includes: &[root],
+	    customize: Customize {
+	      ..Default::default()
+	    },
```

Comment:
Depending on your desired level of clarity, this could also be written as 
```
    includes: &[root],
    customize: Default::default(),
}
```
or
```
    includes: &[root],
    ..Default::default(),
}
```

but it's not necessarily bad to be explicit here that this is the type being used

Reply:
Ah, that was cut and paste, interesting to understand what it is doing